### ISSUE TITLE
DJANGO_CELERY_BEAT_TZ_AWARE conflicts with setting beat_cron_starting_deadline of celery

### ISSUE THREAD

**Issue by cy9747** [celery/django-celery-beat#809]
### Summary:

[beat_cron_starting_deadline](https://docs.celeryq.dev/en/stable/userguide/configuration.html#beat-cron-starting-deadline) is a celery configuration variable.

beat_cron_starting_deadline is used in the [is_due](https://github.com/celery/celery/blob/ad882dc991e46b46c1a195771a897de7127f1e6b/celery/schedules.py#L639C5-L681C28) function of the crontab class, which celery beat's class [TzAwareCrontab](https://github.com/celery/django-celery-beat/blob/8b3992665059172e573ae751851dc2e207761afc/django_celery_beat/tzcrontab.py#L10) inherits from, but overwrites is_due so that this setting does not work.

* Celery Version: 5.4.0
* Celery-Beat Version:  2.7.0

### Exact steps to reproduce the issue:
1. set beat_cron_starting_deadline = 300
2. set DJANGO_CELERY_BEAT_TZ_AWARE = True
3. beat_cron_starting_deadline not work

**Comment by cy9747** [comment]
Is it a good idea to use super?

before

```
def is_due(self, last_run_at):
        last_run_at = last_run_at.astimezone(self.tz)

        rem_delta = self.remaining_estimate(last_run_at)
        rem = max(rem_delta.total_seconds(), 0)
        due = rem == 0
        if due:
            rem_delta = self.remaining_estimate(self.now())
            rem = max(rem_delta.total_seconds(), 0)
        return schedstate(due, rem)
```

after

```
def is_due(self, last_run_at):
    last_run_at = last_run_at.astimezone(self.tz)

    return super().is_due(last_run_at)
```

**Comment by auvipy** [comment]
can you open a PR with failing test case so that we can re produce this?

**Comment by smiling-watermelon** [comment]
@auvipy, as I understand, this should cover what they're talking about.

@cy9747, can you check, and comment/edit if it's not quite right?

**Comment by auvipy** [comment]
thanks for the test, and using super in django model method call is not a bad idea

**Comment by smiling-watermelon** [comment]
BTW, from personal experience the newly updated crontab exclude also _may_ contribute to this problem. (Running code from main branch)

For example, if the task has a schedule like:
`0 5 * * *`, then at hours that are not `5` (either in UTC, or local time, depending on the settings), then this task will be excluded, and thus - will only have a chance to be ran if the `beat_cron_starting_deadline` < 3600 (1 hour).

More concrete example, I've set up a task to run at `0 0 * * *` localtime, at the moment it was 11:07 UTC, with a suffciently large `beat_cron_starting_deadline` (or it being `None`), I'd expect task (that was, for example, ran 2 days ago), I'd expect it to get another run at 11:07 UTC, but it didn't happen, since the task was excluded.

By setting my task to ran at `0 11 * * *`, it did run at 11:07+, so, this is clearly the result of the exclusion by hour.

I don't think that's much of a problem for any real application, but perhaps there should be a way to disable this optimisation, just in case if someone was abusing this specific property of crontab schedules with `last_run_at` property set to previous periods.
(Or, at the very least a mention of that in release notes)

**Comment by auvipy** [comment]
Would you mind fixing that edge case without any regression?

**Comment by smiling-watermelon** [comment]
Sorry, but I can't :(

I'm quite tight on time, and I don't see a straightforward solution to this problem, since it's on the contrary to the optimization that was introduced.


On-Off switch would be nice, of course, but I believe that the comunity + dev team should have a say on that one, not a single person.

**Comment by auvipy** [comment]
Yes no prob. I will ask the  community to share their take on this.

**Comment by auvipy** [comment]
@alirafiei75 since you are on fire with contribution spree, it would be great if you can take a look into this issue as well.

**Comment by alirafiei75** [comment]
> [@alirafiei75](https://github.com/alirafiei75) since you are on fire with contribution spree, it would be great if you can take a look into this issue as well.

I will check it in a couple of days
